synapse/handlers/initial_sync.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, hs):
         self.state = hs.get_state_handler()
         self.clock = hs.get_clock()
         self.validator = EventValidator()
-        self.snapshot_cache = SnapshotCache()
+        self.snapshot_cache = ResponseCache(hs, "initial_sync_cache")
```

Comment:
I'm not convinced that SnapshotCache and ResponseCache are doing the same thing. In particular, SnapshotCache keeps its results for 5 minutes.

I'd suggest keeping the async changes and the replacement of SnapshotCache in separate PRs?

Reply:
I guess, though this brings it into line with sync requests. We can add `timeout_ms=5*60*1000` to get the functionality back

Reply:
https://github.com/matrix-org/synapse/pull/6506